### PR TITLE
REF: use unpack_zerodim_and_defer on EA methods

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -1181,10 +1181,6 @@ def convert_values(param):
                     ovalues = [param] * len(self)
                 return ovalues
 
-            if isinstance(other, (ABCSeries, ABCIndexClass)):
-                # rely on pandas to unbox and dispatch to us
-                return NotImplemented
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/base.py:1186]: Personally, I wouldn't use the decorator here. This class is meant to be used by external EA authors (and in that way also serves somewhat as documentation), and it is not used internally (except for tests). 
I find the "easy understandability" in this case more important than the small code duplication (so it is clear to EA authors this method is returning NotImplemented for pandas objects.

And you can add ABCDataFrame to the check to make it correct (and we should have a test to catch this ..)

**Review by jorisvandenbossche — COMMENTED** [review]
Apart from the one comment about the ScalarOps mixin, looks good.

We have `test_direct_arith_with_series_returns_not_implemented` tests in the base extensions. Could you add `DataFrame` in that test as well?

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (jreback) [reply pandas/core/arrays/base.py:1186]: I think its fine to make this change. why have extra-boilerplate code

Comment (jorisvandenbossche) [reply pandas/core/arrays/base.py:1186]: > why have extra-boilerplate code

For the reason I gave in the above comment

**Comment by jreback** [comment]
@jbrockmendel I am +1 on this, can you merge master. If @jorisvandenbossche object to the array/base.py impl then ok but otherwise lgtm.

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/arrays/base.py:1186]: Can we simply remove it here? (or actually give arguments for why you want to change it)
This is basically a toy/helper implementation that is not used internally. 

(I am completely fine with the rest of the PR)

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
We have `test_direct_arith_with_series_returns_not_implemented` tests in the base extensions. Can you add `DataFrame` in that test as well?

**Comment by jreback** [comment]
can u rebase and add the suggested test

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (jreback) [reply pandas/core/arrays/base.py:1244]: @jbrockmendel I am ok with this here, but I suppose @jorisvandenbossche is stronger here, I guess revert this part.; though this makes things a bit inconsistent across pandas which is odd

**Comment by jbrockmendel** [comment]
Mothballing

**Comment by jorisvandenbossche** [comment]
I reopened this because I think it are useful changes. I only changed it back in the base implementation (which is not used in pandas), as I suggested above. 
@jbrockmendel are you fine with merging it like this?

**Comment by jbrockmendel** [comment]
@jorisvandenbossche i just rebased and force-pushed, might have undone the commit you pushed.  feel free to re-do that

**Comment by jorisvandenbossche** [comment]
I forced pushed again my branch, but it still needs a merge of master.

**Comment by jorisvandenbossche** [comment]
And also merged master

**Comment by jreback** [comment]
lgtm. @jorisvandenbossche merge when ready. (or ping and i will)

**Pull request merged**